lib/miq_automation_engine/engine/miq_ae_method.rb

```diff
@@ -104,13 +115,21 @@ def self.invoke_external(cmd, workspace, serialize_workspace = false)
         end
         rc  = status.exitstatus
         msg = "Method exited with rc=#{verbose_rc(rc)}"
+        method_pid = nil
+        threads = []
       rescue => err
         $miq_ae_logger.error("Method exec failed because (#{err.class}:#{err.message})")
         rc = MIQ_ABORT
         msg = "Method execution failed"
+      ensure
+        if method_pid
+          $miq_ae_logger.error("Terminating non responsive method with pid #{method_pid.inspect}")
+          Process.kill("TERM", method_pid) rescue nil
```

Comment:
I'm not concerned about them returning, I'm concerned about passing in a pid that is no longer valid and it blowing up.